frontend/views/components/Modal/ModalMixins.js

```diff
@@ -13,38 +12,17 @@ const modaMixins = {
     ModalClose
   },
   mounted () {
-    sbp('okTurtles.events/on', CLOSE_MODAL, this.hide)
-    keyboardEvent = window.addEventListener('keyup', (e) => {
-      if (e.key === 'Escape') {
-        this.close()
-      }
-    })
+    sbp('okTurtles.events/on', CLOSE_MODAL, this.close)
   },
   beforeDestroy () {
-    sbp('okTurtles.events/emit', UNLOAD_MODAL)
-    window.removeEventListener('keyup', keyboardEvent)
+    sbp('okTurtles.events/off', CLOSE_MODAL)
   },
   methods: {
     close (e) {
-      sbp('okTurtles.events/emit', CLOSE_MODAL)
-    },
-    hide () {
-      // Only the last open modal should close
-      // TODO: find a more eleguant way to identify the current instance
-      // Maybe saving the name in the modal ?
-      const instance = this.$parent.$options._componentTag
-      const query = this.$route.query
-      if (query) {
-        const subcontent = query.subcontent
-        if (subcontent) {
-          if (subcontent === instance) this.isActive = false
-          // Don't do anything if it's not the last modal
-          return false
-        }
-      }
-      // Remove event only if it's the last modal
-      sbp('okTurtles.events/off', CLOSE_MODAL)
       this.isActive = false
+      setTimeout(() => {
+        sbp('okTurtles.events/emit', UNLOAD_MODAL)
+      }, 300) // Timeout is necessary to let the animation finish on the modal
```

Comment:
Can we avoid the use of magic numbers please? It's not guaranteed that this will be 300 msec on every device, and this could lead to bugs happening at random.

There needs to be a way to definitively know when an animation has finished. Any good animation library is capable of doing that.

Here are some potentially useful resources/libraries:

- https://extralogical.net/articles/firmin-javascript-animation.html
- https://github.com/bendc/animateplus
- https://github.com/juliangarnier/anime
- http://velocityjs.org/

I have not compared those approaches, but one of them probably fits the criteria of "simplest", "uses css transitions", "smallest", and "zero dependencies" (or if not, then some custom combination of them would fit that description)

Reply:
I know already all this. And I took the decision to use this method to simplify the code overall. But only someone with experience on this kind of problem can understand this trade off.

Reply:
I will implement the solution you ask for

Reply:
> The complexity of the code is largely increased for no reason.

@pieer Well hold on, you may be right and I may be the one who is mistaken here. The solution is usually the simplest one, and we want to avoid unnecessary complexity.

You are right that in terms of animation you are more experienced than I am in these matters, so I'll defer to your judgement.

However, I'm pretty sure there's gotta be a way to at least avoid magic numbers in the Cypress tests. Can you check to see if that is true? Can you look to see if there a way to avoid telling Cypress to wait `closeModalDelay` and instead wait for some specific element/selector to appear / become valid?

Reply:
All fixed and simplified (with the trade off of not using sbp for the closing action but only for the unloading which force us to add few element like ref attribute on each modal we create)